hw11/main.cpp

```diff
@@ -0,0 +1,139 @@
+#include <iostream>
+
+const int ROWS = 3;
+const int COLUMNS = 3;
+
+enum SortingDirection { ascending, descending };
+
+bool find(int arr_2d[ROWS][COLUMNS], int toFind) {
+    for (int i = 0; i < ROWS; ++i) {
+        for (int j = 0; j < COLUMNS; ++j) {
+            if (arr_2d[i][j] == toFind) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool isSorted(const int* arr, int size, SortingDirection direction) {
+    if (size <= 1) {
+        return true;
+    }
+
+    if (direction == ascending) {
+        for (int i = 0; i < size - 1; ++i) {
+            if (arr[i] > arr[i + 1]) {
+                return false;
+            }
+        }
+    } else if (direction == descending) {
+        for (int i = 0; i < size - 1; ++i) {
+            if (arr[i] < arr[i + 1]) {
+                return false;
+            }
+        }
+    } else {
+        return false;
+    }
+
+    return true;
+}
+
+void traverseTopDownRightLeftByColumns(int arr_2d[ROWS][COLUMNS]) {
+    for (int col = COLUMNS - 1; col >= 0; --col) {
+        if (col % 2 == 0) {
+            for (int row = 0; row < ROWS; ++row) {
+                std::cout << arr_2d[row][col] << " ";
+            }
+        } else {
+            for (int row = 0; row < ROWS; ++row) {
+                std::cout << arr_2d[row][col] << " ";
+            }
+        }
+    }
+    std::cout << std::endl;
+}
```

Comment:
it doesn't look like a picture from the task

Reply:
Debugging I realized that the 2nd func was incorrect, I also made improvements to the first func